File: note-making-frontend/src/app/services/note.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { Note } from "../models/note.model";

@Injectable({ providedIn: 'root'})
export class NoteService {

    private baseUrl = 'http://localhost:8080/api/note';

    constructor(private http: HttpClient){}

    getAllNotes(): Observable<Note[]>{
        return this.http.get<Note[]>(this.baseUrl);
    }

    createNote(note: Note): Observable<Note>{
        return this.http.post<Note>(this.baseUrl, note);
    }

    updateNote(id: number, note: Note): Observable<Note>{
        return this.http.put<Note>(`${this.baseUrl}/${id}`, note);
    }

    updateNoteStatus(id: number): Observable<Note>{
        return this.http.put<Note>(`${this.baseUrl}/${id}/update-status`, {});
    }

    deleteNote(id: number): Observable<any>{
        return this.http.delete(`${this.baseUrl}/${id}`);
    }

    searchNotes(keyword: string): Observable<Note[]> {
        return this.http.get<Note[]>(`${this.baseUrl}/search?keyword=${keyword}`);
    }

}

File: NoteMakingApplication/src/main/java/com/DXCTrainingProject/NoteMakingApplication/Service/NoteService.java
package com.DXCTrainingProject.NoteMakingApplication.Service;

import com.DXCTrainingProject.NoteMakingApplication.DTO.NoteDTO;
import com.DXCTrainingProject.NoteMakingApplication.DTO.NoteResponseDTO;

import java.util.List;

public interface NoteService {

    NoteResponseDTO createNote(NoteDTO noteDTO);
    List<NoteResponseDTO> getAllNotes();
    NoteResponseDTO updateNote(Long noteId, NoteDTO noteDTO);
    NoteResponseDTO updateNoteStatus(Long noteId);
    String deleteNote(Long noteId);
    List<NoteResponseDTO> searchNotes(String keyword);
    List<NoteResponseDTO> getAllNotesOfUser();
}
